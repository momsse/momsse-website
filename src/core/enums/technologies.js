export default {
  AngularJS: {
    name: 'AngularJS',
    link: 'https://angular.io/',
  },
  Bootstrap: {
    name: 'Bootstrap',
    link: 'https://getbootstrap.com/',
  },
  EJB: {
    name: 'EJB',
    link: 'https://fr.wikipedia.org/wiki/Enterprise_JavaBeans',
  },
  Express: {
    name: 'Express',
    link: 'http://expressjs.com',
  },
  Flex: {
    name: 'Flex',
    link: 'http://flex.apache.org/',
  },
  Glassfish: {
    name: 'Glassfish',
    link:
      'http://www.oracle.com/technetwork/middleware/glassfish/overview/index.html',
  },
  Hibernate: {
    name: 'Hibernate',
    link: 'http://hibernate.org/',
  },
  iBatis: {
    name: 'iBatis',
    link: 'http://ibatis.apache.org/',
  },
  IndexedDB: {
    name: 'IndexedDB',
    link: 'https://developer.mozilla.org/fr/docs/Web/API/API_IndexedDB',
  },
  Java: {
    name: 'Java',
    link: 'https://www.oracle.com/fr/java/',
  },
  Javascript: {
    name: 'Javascript',
    link: 'https://developer.mozilla.org/fr/docs/Web/JavaScript',
  },
  JBoss: {
    name: 'JBoss',
    link: 'http://www.jboss.org/',
  },
  Jersey: {
    name: 'Jersey',
    link: 'https://jersey.github.io/',
  },
  JSF: {
    name: 'JSF',
    link: 'http://www.oracle.com/technetwork/java/javaee/javaserverfaces-139869.html',
  },
  MongoDB: {
    name: 'MongoDB',
    link: 'https://www.mongodb.com/',
  },
  NodeJS: {
    name: 'NodeJS',
    link: 'https://nodejs.org/en/',
  },
  Oracle: {
    name: 'Oracle',
    link: 'https://www.oracle.com/fr/database/index.html',
  },
  PostgreSQL: {
    name: 'PostgreSQL',
    link: 'https://www.postgresql.org/',
  },
  RabbitMQ: {
    name: 'RabbitMQ',
    link: 'https://www.rabbitmq.com/'
  },
  ReactJS: {
    name: 'ReactJS',
    link: 'https://reactjs.org/',
  },
  SocketIO: {
    name: 'SocketIO',
    link: 'https://socket.io/',
  },
  Spring: {
    name: 'Spring',
    link: 'https://spring.io/',
  },
  SpringBatch: {
    name: 'SpringBatch',
    link: 'https://spring.io/projects/spring-batch',
  },
  Struts: {
    name: 'Struts',
    link: 'https://struts.apache.org/',
  },
  Tomcat: {
    name: 'Tomcat',
    link: 'http://tomcat.apache.org/',
  },
  Webpack: {
    name: 'Webpack',
    link: 'https://webpack.js.org/',
  },
  Wicket: {
    name: 'Wicket',
    link: 'https://wicket.apache.org/',
  },
}
