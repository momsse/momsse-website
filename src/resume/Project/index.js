import React, { PropTypes } from 'react'
import {
  Container,
  Technology,
  TechnologyLink,
  Title,
} from './styledComponents'

const Project = ({ title, abstract, technologies }) => (
  <Container>
    <p>
      {title && <Title>{title} :</Title>} {abstract}
    </p>

    {technologies && technologies.map(technology => (
      <Technology key={technology.name}>
        <TechnologyLink href={technology.link}>
          {technology.name}
          {technology.version ? ` ${technology.version}` : ''}
        </TechnologyLink>
      </Technology>
    ))}
  </Container>
)

Project.propTypes = {
  title: PropTypes.string,
  abstract: PropTypes.string,
  technologies: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      link: PropTypes.string,
    })
  ),
}
Project.defaultProps = {}

export default Project
