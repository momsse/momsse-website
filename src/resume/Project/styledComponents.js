import styled from 'styled-components';

export const Container = styled.div`
  font-size: 0.75rem;
`;

export const Title = styled.span`
  font-weight: bold;
`;

export const Technology = styled.span`
  &::after {
    content: ' ';
  }
`

export const TechnologyLink = styled.a.attrs({ target: '_blank' })`
  color: #86994d;
  text-decoration: none;
  @media print {
    color: darken(#86994d, 25%)
  }

  &:hover {
    text-decoration: underline;
  }
`;
