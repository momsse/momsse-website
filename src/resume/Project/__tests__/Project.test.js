import React from 'react'
import renderer from 'react-test-renderer'

import technologies from '../../../core/enums/technologies'
import Project from '../index'

describe('<Project />', () => {
  it('should render successfully', () => {
    const tree = renderer
      .create(
        <Project
          title="Sauron (real time driver and booking manager)"
          abstract="initialization de la stack et developpement de la web app de gestion de la flotte chauffeurs."
          technologies={[
            technologies.Javascript,
            technologies.NodeJS,
            technologies.ReactJS,
            technologies.Webpack,
            technologies.Bootstrap,
            technologies.SocketIO,
            technologies.IndexedDB,
            technologies.MongoDB,
          ]}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
