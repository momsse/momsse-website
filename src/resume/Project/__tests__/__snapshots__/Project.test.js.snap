// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Project /> should render successfully 1`] = `
.c0 {
  font-size: 0.75rem;
}

.c1 {
  font-weight: bold;
}

.c2::after {
  content: ' ';
}

.c3 {
  color: #86994d;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
  className="c0"
>
  <p>
    <span
      className="c1"
    >
      Sauron (real time driver and booking manager)
       :
    </span>
     
    initialization de la stack et developpement de la web app de gestion de la flotte chauffeurs.
  </p>
  <span
    className="c2"
  >
    <a
      className="c3"
      href="https://developer.mozilla.org/fr/docs/Web/JavaScript"
      target="_blank"
    >
      Javascript
      
    </a>
  </span>
  <span
    className="c2"
  >
    <a
      className="c3"
      href="https://nodejs.org/en/"
      target="_blank"
    >
      NodeJS
      
    </a>
  </span>
  <span
    className="c2"
  >
    <a
      className="c3"
      href="https://reactjs.org/"
      target="_blank"
    >
      ReactJS
      
    </a>
  </span>
  <span
    className="c2"
  >
    <a
      className="c3"
      href="https://webpack.js.org/"
      target="_blank"
    >
      Webpack
      
    </a>
  </span>
  <span
    className="c2"
  >
    <a
      className="c3"
      href="https://getbootstrap.com/"
      target="_blank"
    >
      Bootstrap
      
    </a>
  </span>
  <span
    className="c2"
  >
    <a
      className="c3"
      href="https://socket.io/"
      target="_blank"
    >
      SocketIO
      
    </a>
  </span>
  <span
    className="c2"
  >
    <a
      className="c3"
      href="https://developer.mozilla.org/fr/docs/Web/API/API_IndexedDB"
      target="_blank"
    >
      IndexedDB
      
    </a>
  </span>
  <span
    className="c2"
  >
    <a
      className="c3"
      href="https://www.mongodb.com/"
      target="_blank"
    >
      MongoDB
      
    </a>
  </span>
</div>
`;
